ALICE3/TableProducer/onTheFlyTOFPID.cxx

```diff
@@ -0,0 +1,279 @@
+
+      // Now we calculate the expected arrival time following certain mass hypotheses
+      // and the (imperfect!) reconstructed track parametrizations
+      float lThisTrackLengthReco_iTOF = -1, lThisTrackLengthReco_oTOF = -1;
+      auto recoTrack = getTrackParCov(track);
+      if (recoTrack.propagateToDCA(pvvtx, dBz))
+        lX_PV = recoTrack.getX();
+      if (!recoTrack.getXatLabR(iTOFRadius, lX_iTOF, dBz, o2::track::DirOutward))
+        lX_iTOF = -100;
+      if (!recoTrack.getXatLabR(oTOFRadius, lX_oTOF, dBz, o2::track::DirOutward))
+        lX_oTOF = -100;
+      if (lX_PV > -99. && lX_iTOF > -99.)
+        lThisTrackLengthReco_iTOF = TrackLength(recoTrack, lX_PV, lX_iTOF, dBz);
+      if (lX_PV > -99. && lX_oTOF > -99.)
+        lThisTrackLengthReco_oTOF = TrackLength(recoTrack, lX_PV, lX_oTOF, dBz);
+
+      // Straight to Nsigma
+      float lDeltaTime_iTOF[5], lNSigma_iTOF[5];
+      float lDeltaTime_oTOF[5], lNSigma_oTOF[5];
+      int lpdg_array[5] = {11, 13, 211, 321, 2212};
+      float lMasses[5];
+
+      if (doQAplots) {
+        float momentum = recoTrack.getP();
+        float innerbeta = (lThisTrackLength_iTOF / (1e+3 * lMeasuredTime_iTOF)) / o2::constants::physics::LightSpeedCm2NS;
+        float outerbeta = (lThisTrackLength_oTOF / (1e+3 * lMeasuredTime_oTOF)) / o2::constants::physics::LightSpeedCm2NS;
+
+        histos.fill(HIST("h2dVelocityVsMomentumInner"), momentum, innerbeta);
+        histos.fill(HIST("h2dVelocityVsMomentumOuter"), momentum, outerbeta);
+      }
+
+      for (int ii = 0; ii < 5; ii++) {
+        lNSigma_iTOF[ii] = -100;
+        lNSigma_oTOF[ii] = -100;
+
+        auto pdgInfoThis = pdg->GetParticle(lpdg_array[ii]);
+        lMasses[ii] = pdgInfoThis->Mass();
+        lDeltaTime_iTOF[ii] = lThisTrackLengthReco_iTOF / Velocity(recoTrack.getP(), lMasses[ii]) - lMeasuredTime_iTOF;
+        lDeltaTime_oTOF[ii] = lThisTrackLengthReco_oTOF / Velocity(recoTrack.getP(), lMasses[ii]) - lMeasuredTime_oTOF;
+
+        // Fixme: assumes dominant resolution effect is the TOF resolution
+        // and not the tracking itself. It's *probably* a fair assumption
+        // but it should be tested further!
+        if (lThisTrackLength_iTOF > 0 && lThisTrackLengthReco_iTOF)
+          lNSigma_iTOF[ii] = lDeltaTime_iTOF[ii] / iTOFTimeReso;
+        if (lThisTrackLength_oTOF > 0 && lThisTrackLengthReco_oTOF)
+          lNSigma_oTOF[ii] = lDeltaTime_oTOF[ii] / oTOFTimeReso;
+      }
+
+      // Sigmas have been fully calculated. Please populate the NSigma helper table (once per track)
+      upgradetof(lNSigma_iTOF[0], lNSigma_iTOF[1], lNSigma_iTOF[2], lNSigma_iTOF[3], lNSigma_iTOF[4],
+                 lNSigma_oTOF[0], lNSigma_oTOF[1], lNSigma_oTOF[2], lNSigma_oTOF[3], lNSigma_oTOF[4]);
+    }
+  }
+};
+
+WorkflowSpec defineDataProcessing(ConfigContext const& cfgc)
+{
+  WorkflowSpec workflow{adaptAnalysisTask<onTheFlyTOFPID>(cfgc)};
+  return workflow;
```

Comment:
This can be one line: `return WorkflowSpec{...`

Reply:
done!